controllers/volumereplication_controller.go

```diff
@@ -128,6 +129,36 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 	r.Log.Info("volume handle", volumeHandle)
 
+	// check if the object is being deleted
+	if instance.GetDeletionTimestamp().IsZero() {
+		if !contains(instance.GetFinalizers(), volumeReplicationFinalizer) {
+			r.Log.Info("finalizer not found for volumeReplication. Adding finalizer")
+			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, volumeReplicationFinalizer)
+			if err := r.Client.Update(context.TODO(), instance); err != nil {
+				r.Log.Error(err, "failed to update volumeReplication object with finalizer", "objectName", instance.Name, "finalizer", volumeReplicationFinalizer)
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		if contains(instance.GetFinalizers(), volumeReplicationFinalizer) {
```

Comment:
general thought as we are dependent on the PVC object and we don't have control over it. what will happen if the PVC is already deleted and now the user tries to delete the volumeReplicaiton?

Reply:
with the current code, the reconcile will keep on failing at `getPVCDataSource` and volumeReplication CR deletion will be stuck.